generator/common-pipeline.go

```diff
@@ -8,27 +8,39 @@ import (
 
 // Generator is the struct to keep the values passed to the trivy step template
 type Generator struct {
-	TrivyPlugin    string
-	ShellPlugin    string
-	Severity       []string
-	IgnoreUnfixed  bool
-	SecurityChecks []string
-	SkipFiles      string
+	TrivyPlugin     string
+	ShellPlugin     string
+	Severity        []string
+	IgnoreUnfixed   bool
+	SecurityChecks  []string
+	SkipFiles       string
+	ShellCheckFiles string
 }
 
 // GenerateTrivyStep takes trivy plugin version and shell plugin version
 // and an io.Writer to generate trivy step configuration. The trivy step is
 // written to the provided io.Writer.
 // It returns error in case write to the io.Writer errors out.
-func GenerateTrivyStep(trivyPlugin, shellPlugin string, w io.Writer) error {
-	generator := Generator{
-		TrivyPlugin:    trivyPlugin,
-		ShellPlugin:    shellPlugin,
-		Severity:       []string{"CRITICAL", "HIGH"},
-		IgnoreUnfixed:  true,
-		SecurityChecks: []string{"config", "secret", "vuln"},
-		SkipFiles:      "cosign.key",
+func GenerateTrivyStep(trivyPlugin, shellPlugin string, severity []string, ignoreUnfixed bool, securityChecks []string, skipFiles, shellCheckFiles string, w io.Writer) error {
```

Comment:
the input should be a struct, instead of having such a long function definition

Reply:
Updated struct as per the review comments..Thanks @sfunkhouser !